net/stun/stun.go

```diff
@@ -175,12 +177,12 @@ func Response(txID TxID, ip net.IP, port uint16) []byte {
 
 	// Attributes (well, one)
 	b = appendU16(b, attrXorMappedAddress)
-	b = appendU16(b, uint16(4+len(ip)))
+	b = appendU16(b, uint16(4+bytLen))
 	b = append(b,
 		0, // unused byte
 		fam)
-	b = appendU16(b, port^0x2112) // first half of magicCookie
-	for i, o := range []byte(ip) {
+	b = appendU16(b, addrPort.Port()^0x2112) // first half of magicCookie
+	for i, o := range addr.AsSlice() {
```

Comment:
this allocates, which is not ideal

you could do:

```
ipa := addr.As16()
for i, o := range ipa[16-bytLen:] {
```

Reply:
Updated!